newsroom/companies/views.py

```diff
@@ -154,14 +156,45 @@ def update_products(updates, company_id):
 
 def update_company(data, _id):
     updates = {k: v for k, v in data.items() if k in ('sections', 'archive_access', 'events_only')}
+
+    embedded_fields = [
+        'video_display', 'audio_display', 'social_media_display', 'images_display', 'sdpermit_display', 'all_display',
+        'social_media_download', 'video_download', 'audio_download', 'images_download', 'sdpermit_download',
+        'all_download'
+    ]
+
+    if 'embedded' in data:
+        embedded_updates = {k: v for k, v in data['embedded'].items() if k in embedded_fields}
+        if embedded_updates:
+            updates['embedded'] = embedded_updates
+
     get_resource_service('companies').patch(_id, updates=updates)
 
 
 @blueprint.route('/companies/<_id>/permissions', methods=['POST'])
 @account_manager_only
 def save_company_permissions(_id):
+    csrf_token = request.headers.get('X-CSRF-Token')
+    expected_csrf_token = session.get('csrf_token')
```

Comment:
This shouldn't be an issue. The dataflow for this is:
1. [client] request csrf token
2. [server] generate csrf token, store in session, return to client
3. [client] send update company permission request
4. [server] remove csrf token from session and check against request

It's a very short lived token, and the client requests the token and uses it straight away. If the user is concurrently updating many company permissions as the same time, this may be an issue, but I find that hardly unlikely.